src/Microsoft.Health.Fhir.Liquid.Converter/DotLiquids/Validate.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.Health.Fhir.Liquid.Converter.Models.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+using RenderException = Microsoft.Health.Fhir.Liquid.Converter.Exceptions.RenderException;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.DotLiquids
+{
+    public class Validate : Block
+    {
+        private static readonly Regex Syntax = R.B(@"^({0}+)\s$", DotLiquid.Liquid.QuotedFragment);
+
+        private string _schemaFileName;
+        private List<object> _validateBlock;
+
+        public override void Initialize(string tagName, string markup, List<string> tokens)
+        {
+            Match syntaxMatch = Syntax.Match(markup);
+            if (syntaxMatch.Success)
+            {
+                NodeList = _validateBlock = new List<object>();
+                var fullSchemaFileName = syntaxMatch.Groups[1].Value;
+                _schemaFileName = fullSchemaFileName.Substring(1, fullSchemaFileName.Length - 2);
+            }
+            else
+            {
+                throw new SyntaxException(Resources.ValidateTagSyntaxError);
+            }
+
+            base.Initialize(tagName, markup, tokens);
+        }
+
+        public override void Render(Context context, TextWriter result)
+        {
+            JSchema validateSchema = LoadValidateSchema(context);
+
+            if (context is JsonContext jsonContext)
+            {
+                jsonContext.ValidateSchemas.Add(validateSchema);
+            }
+
+            using StringWriter writer = new StringWriter();
+
+            context.Stack(() =>
+            {
+                RenderAll(_validateBlock, context, writer);
+            });
+
+            // Enclose content in validate tag must be a valid JSON format
+            var validateContent = writer.ToString();
+            JObject validateObject;
+            try
+            {
+                validateObject = JObject.Parse(validateContent);
+            }
+            catch (JsonException ex)
+            {
+                throw new RenderException(FhirConverterErrorCode.TemplateRenderingError, "Validate content should be JSON format: " + ex.Message);
+            }
+
+            if (!validateObject.IsValid(validateSchema, out IList<string> messages))
+            {
+                throw new RenderException(FhirConverterErrorCode.InvalidValidateBlockContent, string.Join(";", messages));
+            }
+
+            result.Write(validateContent);
+        }
+
+        private JSchema LoadValidateSchema(Context context)
+        {
+            IFhirConverterTemplateFileSystem fileSystem = context.Registers["file_system"] as IFhirConverterTemplateFileSystem;
+
+            if (fileSystem == null)
+            {
+                throw new RenderException(FhirConverterErrorCode.NullTemplateProvider, Resources.NullTemplateProvider);
+            }
+
+            JsonContentDocument jSchemaDocument = fileSystem.GetTemplate(_schemaFileName)?.Root as JsonContentDocument;
```

Comment:
using function `GetTemplate(context, _schemaFileName)` if you decide to use single quotation for template such as `'xxx.json'`.